# missing-test-assertion
added in: 1.6.0 <span style={{color: 'orange'}}>warning</span>

Warns that there is no assertion in the test.

By default the rule checks for `expect`, `expectLater`, all `expectAsync...` variants and `fail` in `test` and `testWidgets` methods.

## Config
Set `include-assertions` (default is none) to include additional assertions.

Set `include-methods` (default is none) to check additional test methods for missing assertions.
```yaml
dart_code_linter:
  ...
  rules:
    ...
    - missing-test-assertion:
        include-assertions:
          - verify
        include-methods:
          - customTest
```
## Example
### Bad:
```dart
test('bad unit test', () {
    // Given
    final a = 1;
    final b = 2;

    // When
    final c = a + 1;
});
```
### Good:
```dart
test('good unit test', () {
    // Given
    final a = 1;
    final b = 2;

    // When
    final c = a + 1;

    // Then : actual assertion
    expect(b, c);
});
```
