# no-object-declaration
added in: 1.6.0 <span style={{color: 'green'}}>style</span>

Warns when a class member is declared with `Object` type.

## Example
### Bad:
```dart
class Test {
    Object data = 1; // LINT

    Object get getter => 1; // LINT

    // LINT
    Object doWork() {
        return;
    }
}
```
### Good:
```dart
class Test {
    int data = 1;

    int get getter => 1;

    void doWork() {
        return;
    }
}
```
